research/delf/delf/python/training/README.md

```diff
@@ -1,29 +1,134 @@
+* Install the [TensorFlow Object Detection API](https://github.com/tensorflow/models/tree/master/research/object_detection) from the cloned TensorFlow Model Garden repository.
+* Install the DELF package.
+
+*Please note that the current installation only works on 64 bits Linux architectures due to the 
+`protoc` binary downloaded by the installation script. If you wish to install the DELF library on
+other architectures please update the [`install_delf.sh`](./install_delf.sh) script by referencing
+the desired `protoc` [binary release](https://github.com/protocolbuffers/protobuf/releases).*
+
+## Download the GLDv2 Training Data
+
+The [GLDv2](https://github.com/cvdfoundation/google-landmark) images are grouped in 3 datasets: TRAIN, INDEX, TEST. Images in each dataset are grouped into `*.tar` files and individually
+referenced in `*.csv`files containing training metadata and licensing information. The number of
+`*.tar` files per dataset is as follows:
+* TRAIN: 500 files.
+* INDEX: 100 files.
+* TEST: 20 files.
+
+To download the GLDv2 images, run the [`download_dataset.sh`](./download_dataset.sh) script like in
+the following example:
+```
+bash download_dataset.sh 500 100 20
+```
+The script takes the following parameters, in order:
+* The number of image files from the TRAIN dataset to download (maximum 500).
+* The number of image files from the INDEX dataset to download (maximum 100).
+* The number of image files from the TEST dataset to download (maximum 20).
+
+The script downloads the GLDv2 images under the following directory structure:
+* gldv2_dataset/
+  * train/ - Contains raw images from the TRAIN dataset.
+  * index/ - Contains raw images from the INDEX dataset.
+  * test/ - Contains raw images from the TEST dataset.
+
+Each of the three folders `gldv2_dataset/train/`, `gldv2_dataset/index/` and `gldv2_dataset/test/`
+contains the following:
+* The downloaded `*.tar` files.
+* The corresponding MD5 checksum files, `*.txt`.
+* The unpacked content of the downloaded files. (*Images are organized in folders and subfolders
+based on the first, second and third character in their file name.*)
+* The CSV files containing training and licensing metadata of the downloaded images.
 
-## Running training
+*Please note that due to the large size of the GLDv2 dataset, the download can take up to 12 hours and up to 1 TB of space disk.* 
 
-Assuming the data was downloaded to `/tmp/gld_tfrecord/`, running the following
-command should start training a model:
+## Prepare the Data for Training
+
+Preparing the data from training consists of creating [TFRecord](https://www.tensorflow.org/tutorials/load_data/tfrecord)
+files from the raw GLDv2 images grouped into TRAIN, VALIDATION and TEST splits. This can be achieved by running the [`build_image_dataset.py`](./build_image_dataset.py) script. Assuming 
+that the GLDv2 images have been downloaded to the `gldv2_dataset` folder, the script can be run as follows:
```

Comment:
can you please make sure to mention that this will produce a training set which contains only the "clean" subset? You could also add a pointer to the paper as a reference of what the "clean" subset means.

Reply:
Sure. I added the mention that TFRecords will be generated only for the "clean" subset  and I added a reference to the paper for more information.